### ISSUE TITLE
XGBRegressor predict value is not eqaul with model.get_booster().predict value

### ISSUE THREAD

**Issue by yangqingqinghealsci** [dmlc/xgboost#5433]
import xgboost as xgb
from xgboost.sklearn import XGBRegressor
import xgboost
import pandas as pd
import numpy as np
from sklearn import datasets
from xgboost import DMatrix
boston_data = datasets.load_boston()
df = pd.DataFrame(boston_data.data,columns=boston_data.feature_names)
df['target'] = pd.Series(boston_data.target)

Y = df["target"]
X = df.drop('target', axis=1)

#### Code using Sklearn Wrapper for XGBoost
model = XGBRegressor(n_estimators = 200, max_depth=3, learning_rate =.05, min_child_weight=4, subsample=0.8, missing=0.0 )

#model = model.fit(X, Y, eval_set = [(X, Y), (X, Y)], eval_metric = 'rmse', verbose=True)
model = model.fit(X, Y)

predictions3 = model.get_booster().predict(DMatrix(X))
print(predictions2[0] , predictions3[0])
print(np.absolute(predictions3-predictions2).sum())

**Comment by yangqingqinghealsci** [comment]
![image](https://user-images.githubusercontent.com/29110899/77151020-14078380-6ad0-11ea-949a-00464335c967.png)

**Comment by trivialfis** [comment]
Set the `missing` value for `DMatrix` to `0.0` please.
